src/utilities/effects.ts

```diff
@@ -4359,4 +4386,39 @@ const effects = {
   },
 };
 
+function replacePaths(modelParameters: ParametersMap | null, simArgs: ArgumentsMap, filenames: any): ArgumentsMap {
+  if (modelParameters === null) {
+    return simArgs;
+  }
+  const result: ArgumentsMap = {};
+  for (const parameterName in modelParameters) {
+    const parameter: Parameter = modelParameters[parameterName];
+    const arg: Argument = simArgs[parameterName];
+    if (arg === undefined) {
+      continue;
+    }
+    result[parameterName] = replacePathsHelper(parameter.schema, arg, filenames);
+  }
+  return result;
+}
+
+function replacePathsHelper(schema: ValueSchema, arg: Argument, filenames: any) {
```

Comment:
Would it be possible to maybe include a few unit tests for this? Also, is there a way to narrow down the type for `filenames`?

Reply:
I pushed a commit to specify the type as `Record<string, string>`. I'll look into writing some unit tests - I may need some guidance, I'll let you know.

Reply:
Added a test (which caught a bug!) - let me know what you think